services/web/server/src/simcore_service_webserver/users_handlers.py

```diff
@@ -109,3 +113,64 @@ async def delete_token(request: web.Request):
         raise web.HTTPNoContent(content_type=MIMETYPE_APPLICATION_JSON)
     except TokenNotFoundError as exc:
         raise web.HTTPNotFound(reason=f"Token for {service_id} not found") from exc
+
+
+# me/notifications -----------------------------------------------------------
+
+async def _get_user_notifications(redis_client: aioredis.Redis, user_id: int):
+    notifs = []
+    notif_wildcard = f'user_id={user_id}:notification_id=*'
+    async for scanned_notification_key in redis_client.scan_iter(match=notif_wildcard):
```

Comment:
Unfortunately this is not fine when dealing with lots of users and notifications.
The Redis command [SCAN](https://redis.io/commands/scan/) you are using is `O(n)` time complexity where `n` is the number of keys you are scanning over. In this implementation this means that each time a user asks for the notifications, Redis has to scan over all the keys it has stored in the database.

A better approach would be to store all notifications for a user in a separate list with [LPUSH](https://redis.io/commands/lpush/). And query the last `10` entries in the list with [LRANGE](https://redis.io/commands/lrange/).
You have the following advantages with this approach:
- notification insertion order is guaranteed
- the key of the list that contains the notifications can just contain the `user_id` without the notification
- at this point if you do not require it, the notification_id can be dropped

Reply:
Done, merci!